File: src/main/java/com/baulsupp/oksocial/authenticator/ValidatedCredentials.java
package com.baulsupp.oksocial.authenticator;

import java.util.Optional;

public class ValidatedCredentials {
  public final Optional<String> username;
  public final Optional<String> clientName;

  public ValidatedCredentials(Optional<String> username, Optional<String> clientName) {
    this.username = username;
    this.clientName = clientName;
  }

  public ValidatedCredentials(String username, String client) {
    this(Optional.ofNullable(username), Optional.ofNullable(client));
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/authenticator/ValidatedCredentials.kt
package com.baulsupp.oksocial.authenticator

import java.util.Optional

class ValidatedCredentials(val username: Optional<String>, val clientName: Optional<String>) {

    constructor(username: String, client: String) : this(Optional.ofNullable<String>(username), Optional.ofNullable<String>(client)) {}
}
